src/library_wasi.js

```diff
@@ -62,6 +62,33 @@ var WasiLibrary = {
     });
     return 0;
   },
+
+  args_sizes_get: function(argc, argv_buf_size) {
+#if MAIN_READS_PARAMS
+    {{{ makeSetValue('argc', 0, 'mainArgs.length', 'i32') }}};
+    var bufSize = 0;
+    mainArgs.forEach(function(arg) {
+      bufSize += arg.length + 1;
+    });
+    {{{ makeSetValue('argv_buf_size', 0, 'bufSize', 'i32') }}};
+#else
+    {{{ makeSetValue('argc', 0, '0', 'i32') }}};
+#endif
+    return 0;
+  },
+
+  args_get: function(argv, argv_buf) {
+#if MAIN_READS_PARAMS
+    var bufSize = 0;
+    mainArgs.forEach(function(arg, i) {
+      var ptr = argv_buf + bufSize;
+      {{{ makeSetValue('argv', 'i * 4', 'ptr', 'i32') }}};
+      writeAsciiToMemory(arg, ptr);
```

Comment:
the other code path uses utf8, why is this ascii? should they both be ascii?

Reply:
I'm just copying from `environ_get` above.

Reply:
I honestly don't know what's right here. Can env vars be non-ascii, and can commandline params?

Reply:
Yes, utf8 should work in both cases.     Lets do that as a followup and include tests.

Reply:
Alternatively if ascii is enough, that would be better for code size. But I really don't know if people expect utf8 in those places...

Reply:
I general people may well want UTF8 in both argv and envp.  Imagine for example a node program that wants to accepts a filename as argv[1].